jplag/src/main/java/de/jplag/JPlag.java

```diff
@@ -67,29 +58,19 @@ public JPlagResult run() throws ExitException {
                     ExitException.NOT_ENOUGH_SUBMISSIONS_ERROR);
         }
 
-        // 3. Compare valid submissions:
+        // Compare valid submissions.
         JPlagResult result = comparisonStrategy.compareSubmissions(submissionSet);
         System.out.println("Total time for comparing submissions: " + TimeUtil.formatDuration(result.getDuration()));
         return result;
     }
 
-    /**
-     * @return the configured language in which the submissions are written.
-     */
-    public Language getLanguage() {
-        return language;
-    }
-
-
     /**
      * This method checks whether the base code directory value is valid.
      */
     private void checkBaseCodeOption() throws ExitException {
-        if (options.hasBaseCode()) {
-            if (!new File(options.getRootDirectoryName()).exists()) {
-                throw new ExitException("Root directory \"" + options.getRootDirectoryName() + "\" doesn't exist!", ExitException.BAD_PARAMETER);
-            }
+        getRootDirectory(); // Performs checks on the root directory.
 
+        if (options.hasBaseCode()) {
             String baseCode = options.getBaseCodeSubmissionName().replace(File.separator, ""); // trim problematic file separators
```

Comment:
The file separator of the basecode submission name is already trimmed here.

Reply:
Yes, and at old line 109 below it gets sneaked back into the options, so it's changed in this function.

Since this code is called from the constructor, all other uses of `options.getBaseCodeSubmissionName()` get the trimmed version, and can compare `File.getName()` with the option value and get a correct answer.  
I just moved this trim call to the option setter itself.

I see now it's not deleted, I agree that's not good.

Reply:
> I see now it's not deleted, I agree that's not good.
Yes, let's delete it here and keep it in the options.